core/src/engine/EngineMultiRobot.cc

```diff
@@ -808,6 +808,15 @@ namespace jiminy
             returnCode = start(xInit, true, false);
         }
 
+        // Now that telemetry has been initialized, check simulation duration.
+        if (tEnd > telemetryRecorder_->getMaximumLogTime())
```

Comment:
You should also check that condition in `step` method.

Reply:
On the one hand, it would be nice to only have it at one place (in `step`), but on the other hand, it is useful to be warn the user that overflow will happen calling simulate method with too large duration. What do you think ?